spec/services/openlibrary_service_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+describe OpenlibraryService do
+  include OpenlibraryMock
+  let(:service) { OpenlibraryService.new }
+  let(:uri) { 'https://openlibrary.org/api/books' }
+
+  describe '#search_by_isbn' do
+    subject { service.search_by_isbn(isbn) }
+
+    context 'when getting a successfuly response' do
+      before do
+        successfuly_book_get(isbn)
+        subject
+      end
+
+      let(:isbn) { '0385472579' }
+
+      it 'makes an external web request' do
+        expect(WebMock)
+          .to have_requested(:get, "#{uri}?bibkeys=ISBN:#{isbn}&format=json&jscmd=data")
+          .once
+      end
+
+      it 'responses with a Hash' do
+        is_expected.to be_a Hash
+      end
+
+      it 'responses with a Hash with expected keys' do
+        expected_keys = %i[isbn title subtitle number_of_pages authors]
+        is_expected.to include(*expected_keys)
+      end
```

Comment:
We can test status code be :ok

Reply:
I haven't got access to the http response outside the service object